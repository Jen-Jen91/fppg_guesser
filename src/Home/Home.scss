.home-container {
  display: flex;
  flex-direction: column;
  align-items: center;

  height: 100vh;
}

.title {
  display: flex;
  align-items: center;
  justify-content: center;

  margin: 0;
  width: 100vw;
  height: 150px;

  background-color: #1f375b;

  color: #ffffff;
}

.players-container {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;

  width: 100vw;
}

.score {
  display: flex;
  align-items: center;
  justify-content: center;

  margin: 0 0 50px 0;
  border: solid 2px #1f375b;
  border-top: 0px;
  border-bottom-right-radius: 10px;
  border-bottom-left-radius: 10px;
  width: 300px;
  height: 50px;

  background-color: #ffffff;

  color: #1f375b;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.5);
}

.results-container {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: #ffffff;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;

  width: 200px;
  height: 200px;

  border: solid 2px #1f375b;
  border-radius: 10px;
}

.results-text {
  color: #1f375b;
  font-size: 1.2rem;
  font-weight: bold;
}

.next-text {
  color: #ffffff;
  margin: 0;
}
