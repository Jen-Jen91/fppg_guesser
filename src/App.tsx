import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Home from "./Home/Home";
import End from "./End/End";

const App = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/end" component={End} />
      </Switch>
    </Router>
  );
};

export default App;
