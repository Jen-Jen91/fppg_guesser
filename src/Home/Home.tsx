import React, { useEffect, useState } from "react";
import { RouteComponentProps, withRouter } from "react-router";
import { APIData, MainPlayerData } from "../Common/Types";
import Player from "../Common/Player/Player";
import Button from "../Common/Button/Button";
import Modal from "../Common/Modal/Modal";
import "./Home.scss";

export interface ExtendedProps extends RouteComponentProps {}

const Home = (props: ExtendedProps) => {
  const [score, setScore] = useState(0);
  const [loading, setLoading] = useState(true);
  const [isCorrect, setIsCorrect] = useState(false);
  const [showResults, setShowResults] = useState(false);
  const [allPlayers, setAllPlayers] = useState<MainPlayerData[]>([]);

  const player: MainPlayerData = {
    firstName: "",
    lastName: "",
    imageUrl: "",
    fppg: 0
  };

  const [playerOne, setPlayerOne] = useState<MainPlayerData>(player);
  const [playerTwo, setPlayerTwo] = useState<MainPlayerData>(player);

  useEffect(() => {
    fetchPlayerData();
  }, []);

  useEffect(() => {
    if (showResults || allPlayers.length <= 1) return;

    let uniqueIndices: number[] = [];

    if (allPlayers.length === 2) {
      uniqueIndices = [0, 1];
    } else {
      let i = uniqueIndices.length;

      while (i < 2) {
        const max = allPlayers.length - 1;
        const random = Math.round(Math.random() * max);

        if (!uniqueIndices.includes(random)) {
          uniqueIndices.push(random);
          i++;
        }
      }
    }

    setPlayerOne(allPlayers[uniqueIndices[0]]);
    setPlayerTwo(allPlayers[uniqueIndices[1]]);
  }, [allPlayers, showResults]);

  const fetchPlayerData = async () => {
    try {
      const resp = await fetch(
        "https://gist.githubusercontent.com/liamjdouglas/bb40ee8721f1a9313c22c6ea0851a105/raw/6b6fc89d55ebe4d9b05c1469349af33651d7e7f1/Player.json"
      );
      const allData: APIData = await resp.json();

      const playersData: MainPlayerData[] = allData.players.map(player => {
        const data: MainPlayerData = {
          firstName: player.first_name || "",
          lastName: player.last_name || "",
          imageUrl: player.images.default.url || "",
          fppg: player.fppg || 0
        };
        return data;
      });

      setAllPlayers(playersData);
      setLoading(false);
    } catch (e) {
      console.error(e);
    }
  };

  const playerGuessed = (fppg: number) => {
    setShowResults(true);

    if (fppg >= playerOne.fppg && fppg >= playerTwo.fppg) {
      setIsCorrect(true);
      updateScore();
    } else {
      setIsCorrect(false);
    }
  };

  const updateScore = () => {
    if (score === 9) {
      setScore(score + 1);
      props.history.push("/end");
    } else {
      setScore(score + 1);
    }
  };

  return (
    <main className="home-container">
      <h1 className="title">
        Guess the player with the highest FanDuel Points Per Game (FPPG)
      </h1>

      <h3 className="score">Score: {score}</h3>

      {loading || allPlayers.length === 0 ? (
        <div className="loader" />
      ) : (
        <section className="players-container">
          <Player
            playerFirstName={playerOne.firstName}
            playerLastName={playerOne.lastName}
            url={playerOne.imageUrl}
            fppg={playerOne.fppg}
            clickPlayer={playerGuessed}
            showFppg={showResults}
          />

          <Player
            playerFirstName={playerTwo.firstName}
            playerLastName={playerTwo.lastName}
            url={playerTwo.imageUrl}
            fppg={playerTwo.fppg}
            clickPlayer={playerGuessed}
            showFppg={showResults}
          />
        </section>
      )}

      {showResults && (
        <Modal>
          <div className="results-container">
            <p className="results-text">{isCorrect ? "Correct!" : "Wrong!"}</p>
            <Button handleClick={() => setShowResults(false)}>
              <p className="next-text">Next</p>
            </Button>
          </div>
        </Modal>
      )}
    </main>
  );
};

export default withRouter(Home);
