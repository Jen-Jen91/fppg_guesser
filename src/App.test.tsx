import React from "react";
import App from "./App";
import Home from "./Home/Home";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";

describe("App", () => {
  test("snapshot test", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("redirects to Home", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );

    expect(wrapper.find(Home)).toHaveLength(1);
  });
});
