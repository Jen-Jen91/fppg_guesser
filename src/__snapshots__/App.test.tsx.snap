// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<h1>
  FPPG Guesser
</h1>
`;
